tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,563 @@
+	masterIp string, testUserAlias string, testUser string, dataCenter string, datastores []string) error {
+	for i := 0; i < len(datastores); i++ {
+		setPermissionForDataStore := govcLoginCmd(vcAddress, vCenterPort) + "govc permissions.set -principal " +
+			testUserAlias + " " +
+			"-propagate=false -role CNS-DATASTORE-" + testUser + " '" + datastores[i] + "'"
+		framework.Logf("Assign datastore level Permissions to user: %s ", setPermissionForDataStore)
+		result, err := sshExec(sshClientConfig, masterIp, setPermissionForDataStore)
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForDataStore, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setSearchlevelPermission method is used to set search level permissions
+func setSearchlevelPermission(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, testUserAlias string, testUser string) error {
+	setSearchLevelPermission := govcLoginCmd(vcAddress, vCenterPort) + "govc permissions.set -principal " + testUserAlias +
+		" -propagate=false -role CNS-SEARCH-AND-SPBM-" + testUser + " /"
+	framework.Logf("Assign search level Permissions to user: %s ", setSearchLevelPermission)
+	result, err := sshExec(sshClientConfig, masterIp, setSearchLevelPermission)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			setSearchLevelPermission, masterIp, err)
+	}
+	return nil
+}
+
+// createCsiVsphereSecret method is used to create csi vsphere secret file
+func createCsiVsphereSecret(client clientset.Interface, ctx context.Context, testUser string,
+	password string, csiNamespace string) {
+	currentSecret, err := client.CoreV1().Secrets(csiNamespace).Get(ctx, configSecret, metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	originalConf := string(currentSecret.Data[vSphereCSIConf])
+	vsphereCfg, err := readConfigFromSecretString(originalConf)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	vsphereCfg.Global.User = testUser
+	vsphereCfg.Global.Password = password
+	modifiedConf, err := writeConfigToSecretString(vsphereCfg)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	ginkgo.By("Updating the secret to reflect new conf credentials")
+	currentSecret.Data[vSphereCSIConf] = []byte(modifiedConf)
+	_, err = client.CoreV1().Secrets(csiNamespace).Update(ctx, currentSecret, metav1.UpdateOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+// restartCSIDriver method restarts the csi driver
+func restartCSIDriver(ctx context.Context, client clientset.Interface, namespace string,
+	csiReplicas int32) (bool, error) {
+	isServiceStopped, err := stopCSIPods(ctx, client)
+	if err != nil {
+		return isServiceStopped, err
+	}
+	isServiceStarted, err := startCSIPods(ctx, client, csiReplicas)
```

Comment:
in `startCSIPods()`,I think we should fetch csi pod count before updating the deployment, can you make that fix as well